### ISSUE TITLE
[Research] Replace Lodash library by JS methods

### ISSUE THREAD

**Issue by nurSaadat** [opensearch-project/OpenSearch-Dashboards#4196]
As mentioned in #3661 Lodash can be replaced in some files by pure JS thanks to the development in JS functionalities.

Lodash methods are currently can be found in 813 files,
[here is the list](https://github.com/opensearch-project/OpenSearch-Dashboards/files/11287626/out.txt)

My approach would be to change the methods package by package, plugin by plugin. To be specific that would include 93 different subtasks listed in [this file](https://github.com/opensearch-project/OpenSearch-Dashboards/files/11287630/subtasks.txt)

I suggest reducing the number of subtasks, as I still need to calculate how many files will be audited in every plugin.

I'd like to use [this conversion table](https://github.com/opensearch-project/OpenSearch-Dashboards/files/11287650/Lodash.Migration.-.Sheet2.pdf) created by research and ChatGPT help (probably needs correction).

I propose to create separate issues for each subtask for discussion and easier navigation.

References to resources used:
https://www.sitepoint.com/lodash-features-replace-es6/
https://blog.bitsrc.io/you-dont-need-lodash-or-how-i-started-loving-javascript-functions-3f45791fa6cd

**Comment by ananzh** [comment]
@nurSaadat Since you have done a lot of research work on this, do you want to take this task? or help creating sub-tasks and take part of them?

**Comment by nurSaadat** [comment]
@ananzh yes, I was trying to tackle this problem awhile ago, and I will be happy to take on this issue and share work with other contributors.

### What is being planned
@Aigerim-ai and I are planning to create sub-tasks based on the plugins and packages where Lodash is present. So far we discussed that we are going to 
* create 86(!) sub issues that are going to cover the whole repo
* create a [Meta] tagged issue to tell the story and list all issues to track them
* create an example PR following and resolving one of 86 issues to guide future contributors

### Contents of Lodash issues
In the issues we are going to mention a plugin/package/directory to work on as well as mention Lodash occurrences in the plugin/package and list files in the issue description. (Like in OUI issues)

**What we expect from contributors working on those issues is**
1. List functions that are going to be replaced 
2. Mention, if some of the functions were already present in replace_lodash.js 
3. Attach PRs, which replace the functions mentioned above
4. Run tests and make sure they are working properly 
 
Right now @Aigerim-ai is working on the [Meta] issue describing the story of the changes

**Comment by shivm29** [comment]
Hey @nurSaadat ! can you assign me this issue? I wanna contribute !!

**Comment by vvavdiya** [comment]
Hi @nurSaadat @ananzh
I would like to contribute into this issue. Can you please assign me this issue.
